src/Featurizers/Structs.h

```diff
@@ -1,245 +1,329 @@
+        value_type const                    Value;
+        std::uint64_t const                 Index;
+
+        // ----------------------------------------------------------------------
+        // |  Public Methods
+        ValueEncoding(value_type value, std::uint64_t index);
+
+        bool operator==(ValueEncoding const &other) const;
+        bool operator!=(ValueEncoding const &other) const;
+    };
+
+    // ----------------------------------------------------------------------
+    // |
+    // |  Public Data
+    // |
+    // ----------------------------------------------------------------------
+    std::uint64_t const                     NumElements;
+    std::vector<ValueEncoding> const        Values;
+
+    // ----------------------------------------------------------------------
+    // |
+    // |  Public Methods
+    // |
+    // ----------------------------------------------------------------------
+    SparseVectorEncoding(std::uint64_t numElements, std::vector<ValueEncoding> values);
+
+    bool operator==(SparseVectorEncoding const &other) const;
+    bool operator!=(SparseVectorEncoding const &other) const;
+};
+
+/////////////////////////////////////////////////////////////////////////
+///  \class         SingleValueSparseVectorEncoding
+///  \brief         Object that contains information that can be used to create
+///                 a sparse vector with only a single value set without creating
+///                 the vector itself.
+///
+template <typename T>
+class SingleValueSparseVectorEncoding {
+public:
+    // ----------------------------------------------------------------------
+    // |
+    // |  Public Types
+    // |
+    // ----------------------------------------------------------------------
+    using value_type                        = T;
+
+    // ----------------------------------------------------------------------
+    // |
+    // |  Public Data
+    // |
+    // ----------------------------------------------------------------------
+    std::uint64_t const                     NumElements;
+    value_type const                        Value;
+    std::uint64_t const                     Index;
+
+    // ----------------------------------------------------------------------
+    // |
+    // |  Public Methods
+    // |
+    // ----------------------------------------------------------------------
+    SingleValueSparseVectorEncoding(std::uint64_t numElements, value_type value, std::uint64_t index);
+
+    bool operator==(SingleValueSparseVectorEncoding const &other) const;
+    bool operator!=(SingleValueSparseVectorEncoding const &other) const;
+};
+
+/////////////////////////////////////////////////////////////////////////
+///  \struct        TFStruct
+///  \brief         Struct to hold return value of count vectorizer,
+///                 consist of <dictid, # of appearances>
+///
+struct TFStruct {
+    std::uint32_t const DictionaryId;                              // dict id
```

Comment:
>DictionaryId [](start = 24, length = 12)

Underscore in the name?

Reply:
This struct will be deprecated in next update